addons/mass_mailing/models/mailing_ab_testing.py

```diff
@@ -0,0 +1,238 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import random
+from ast import literal_eval
+from datetime import timedelta
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+from odoo.osv import expression
+
+
+class MassMailingTestingCampaign(models.Model):
+    """
+    This models allows to manage A/B Testing Campaigns
+    """
+    _name = 'mailing.ab.testing'
+    _description = "A/B Testing Campaign"
+    _inherit = ['mail.thread', 'mail.activity.mixin', 'mail.render.mixin']
+
+    def _get_default_mailing_domain(self):
```

Comment:
I see you copy/pasted from mailing.mailing.
Be careful that mass_mailing_sms overrides that to add its own blacklist mechanic. 

(Also a bit strange that this is above field definition).